core/domain/base_model_validators.py

```diff
@@ -84,12 +89,75 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
-        """
+
+        Raises:
+            Exception. This class was used instead of
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
+        """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'When fetching instances of UserSettingsModel, please use ' +
+                'UserSettingsModelFetcherDetails instead of ' +
+                'ExternalModelFetcherDetails')
         self.field_name = field_name
         self.model_class = model_class
         self.model_ids = model_ids
 
 
+class UserSettingsModelFetcherDetails(python_utils.OBJECT):
+    """Value object providing ids to fetch the user settings model."""
+
+    def __init__(
+            self, field_name, model_ids,
+            may_contain_system_ids, may_contain_pseudonymous_ids):
+        """Initializes the UserSettingsModelFetcherDetails domain object.
+
+        Args:
+            field_name: str. A specific name used as an identifier by the
+                storage model which is used to identify the user settings model
+                reference. For example: 'committer_id': UserSettingsModel,
```

Comment:
For example: `'committer_id': UserSettingsModel` means that committer_id is a field which contains a user_id used to identify the ...

Reply:
Done.

Reply:
Not quite. Drop the comma after UserSettingsModel and put backticks around `'committer_id': UserSettingsModel` to show that it is a piece of code.

Reply:
Done.